File: submissions/iqbal/problem3.cpp
#include<iostream>
#include<math.h>
using namespace std;
int main()
{
 int t;
 cin>>t;
 cout<<endl;
 long unsigned int a[2][t],v[t];
 long double m[t];
 for(int i=0;i<t;i++)
  {
    for(int k=0;k<2;k++)
    cin>>a[k][i];
   cout<<endl;
  }
for(int j=0;j<t;j++)
  {
   m[j]=pow(a[0][j],a[1][j]);
   v[j]=m[j]/10;
   v[j]*=10;
   cout<<m[j]-v[j]<<endl;
  }
 return 0;
}

File: submissions/Gunjan4542/problem3.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


import java.io.BufferedReader;
import java.io.InputStreamReader;

/**
 *
 * @author tcstcs
 */
public class problem3 {
    static BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
    public static void main(String[]args)throws Exception{
        int t=(int)getInt();
        for(int c=0;c<t;c++){
            int arr[]=getIntArr();
            int ld[]=new int[6];
            for(int i=0;i<=5;i++){
                ld[i]=(int)Math.pow(arr[0],i)%10;
            }
            System.out.println(last(ld,arr[1]));
        }
    }
    static int last(int ld[],int num){
        if(num/5==0){
            return ld[num];
        }
        else{
            return last(ld,num/5+num%5);
        }
    }
    
 
    static long getInt()throws Exception{
        long t=Long.parseLong(br.readLine());
        return  t;
    }
    static int [] geIntArr()throws Exception{
        String s[]=br.readLine().split(" ");
        int arr[]=new int[s.length];
        for(int i=0;i<s.length;i++){
            arr[i]=Integer.parseInt(s[i]);
        }
        return arr;
    }
     static int [] getIntArr()throws Exception{
        String s[]=br.readLine().split(" ");
        int arr[]=new int[s.length];
        for(int i=0;i<s.length;i++){
            arr[i]=Integer.parseInt(s[i]);
        }
        return arr;
    }
     static long [] getLongArr()throws Exception{
        String s[]=br.readLine().split(" ");
        long arr[]=new long[s.length];
        for(int i=0;i<s.length;i++){
            arr[i]=Long.parseLong(br.readLine());
        }
        return arr;
    }
    
}
